### PR TITLE
feat(figma-utils): sort variables when parsing

### PR DESCRIPTION
**Pull request by larsrickert**
## Description

Sort variables by:
1. alphabetically
2. if two variables only differ in a number (e.g. `color-100` and `color-1000`), sort them numerically so that 100 is sorted before 1000

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] All changes are documented in the documentation app (folder `apps/docs`)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/figma-utils/src/utils/parse.ts [hunk 1]
```diff
@@ -61,6 +61,25 @@ export const parseFigmaVariables = (
...
+    // sort them by number instead of alphabetically so e.g. 100 is sorted before 1000
+    data.variables = Object.keys(data.variables)
+      .sort((a, b) => {
+        const aAsNumber = numberRegex.exec(a)?.[0] ?? "";
+        const bAsNumber = numberRegex.exec(b)?.[0] ?? "";
+        const isSameBaseName = a.replace(aAsNumber, "") === b.replace(bAsNumber, "");
+
+        if (aAsNumber && bAsNumber && isSameBaseName) return +aAsNumber - +bAsNumber;
+        return a.localeCompare(b);
+      })
```
Comment (BoppLi) [inline packages/figma-utils/src/utils/parse.ts:78]: if we have many, we would do the transformation operation many many times. suggestion:
```suggestion
      .map((key) => {
        const counter = numberRegex.exec(key)?.[0] ?? "";
        return {
          key,
          counter,
          base: key.replace(counter, ""),
        };
      })
      .sort((a, b) => {
        const isSameBaseName = a.base === b.base;

        if (a.counter && b.counter && isSameBaseName) return +a.counter - +b.counter;
        return a.key.localeCompare(b.key);
      })
      .map(bundle => bundle.key)
```

File: packages/figma-utils/src/utils/parse.ts [hunk 2]
```diff
@@ -61,6 +61,25 @@ export const parseFigmaVariables = (
...
     if (data.modeName === DEFAULT_MODE_NAME) delete data.modeName;
+
+    const numberRegex = /\d+/;
+
+    // sort variables by name
+    // for variables with the same name that just end with a different number (e.g. my-var-100 and my-var-200)
+    // sort them by number instead of alphabetically so e.g. 100 is sorted before 1000
+    data.variables = Object.keys(data.variables)
+      .sort((a, b) => {
+        const aAsNumber = numberRegex.exec(a)?.[0] ?? "";
```
[the remark this answers is not in the document]
Comment (larsrickert) [reply packages/figma-utils/src/utils/parse.ts:72]: Imo this should also be sorted numericaly, so e.g. "foo-200-bar", "foo-100-bar" should be sorted as
1. foo-100-bar
2. foo-200-bar

File: packages/figma-utils/src/utils/parse.ts (hunk 1 above)
Comment (larsrickert) [reply packages/figma-utils/src/utils/parse.ts]: Done

Comment (BoppLi) [reply packages/figma-utils/src/utils/parse.ts]: Nice!!! I like your change! 😊

**Review by BoppLi — APPROVED** [review]

**Pull request merged**